packages/angular/src/lib/flag.directive.ts

```diff
@@ -1,39 +1,58 @@
-import {
-  Directive,
-  Input,
-  OnInit,
-  TemplateRef,
-  ViewContainerRef,
-} from "@angular/core";
-
+import { Directive, Input, OnDestroy, OnInit, TemplateRef, ViewContainerRef } from "@angular/core";
+import { Subject, takeUntil, map, distinctUntilChanged, ReplaySubject, switchMap } from "rxjs";
 import { AbbyService } from "./abby.service";
 
 @Directive({
   selector: "[featureFlag]",
 })
-export class AbbyFlag implements OnInit {
-  @Input() featureFlag: string;
+export class AbbyFlag implements OnInit, OnDestroy {
+  @Input()
+  set featureFlag(featureFlag: string) {
+    // ensure featureFlag is a string to quit gracefully
+    if(typeof featureFlag !== 'string') {
+      console.warn(`Expected a string as featureFlag. Got ${featureFlag}`);
+      return;
+    }
+
+    this.currentFlag$.next(featureFlag);
+  }
+
+  private currentFlag$ = new ReplaySubject<string>(1);
+  private _destroy$ = new Subject<void>();
 
   constructor(
     private readonly abby: AbbyService,
-
     private _viewContainer: ViewContainerRef,
     private _templateRef: TemplateRef<any>
   ) {}
 
   ngOnInit(): void {
-    let flagName: string;
-    if (this.featureFlag.startsWith("!"))
-      flagName = this.featureFlag.substring(1);
-    else flagName = this.featureFlag;
-    this.abby.getFeatureFlagValue(flagName).subscribe((value) => {
-      this._viewContainer.clear();
-      if (
-        (value && this.featureFlag[0] != "!") ||
-        (this.featureFlag[0] == "!" && !value)
-      ) {
-        this._viewContainer.createEmbeddedView(this._templateRef);
-      }
-    });
+    this.currentFlag$
+      .pipe(
+        switchMap((flagName) => {
+          return this.abby
+            .getFeatureFlagValue(flagName.startsWith("!") ? flagName.slice(1) : flagName)
```

Comment:
why do we need to do it here? this should be part of the service? what if any1 else calls `getFeatureFlagValue` with a flagName starting with `!`?